zrml/lsd-lmsr/src/constants.rs

```diff
@@ -0,0 +1,36 @@
+use crate::types::Timespan;
+use substrate_fixed::{types::extra::U24, types::extra::U32, FixedU32};
+
+// --- Default configuration for EmaConfig struct ---
+/// Default short EMA period (in seconds)
+/// One hour.
+pub const EMA_SHORT: Timespan = Timespan::Hours(1);
+
+/// Default long EMA period (in seconds)
+/// Six hours.
+pub const EMA_LONG: Timespan = Timespan::Hours(6);
+
+/// Default smoothing factor for EMA calculation
+/// 2.0
+pub const SMOOTHING: FixedU32<U24> = <FixedU32<U24>>::from_bits(0x0200_0000);
```

Comment:
It is a pity `from_num` is not a constant function.

Can you answer me a question? The default substrate `FixedU128` structure has 18 digits and an internal `2_000_000_000_000_000_000` decimal actually means `2.0`. So does `U24` here means the number of decimal digits (fractions)?

Given that:

- `u32` can only hold a maximum of `4294967295` decimal.
- `0x0200_0000` hex is `33554432` decimal but in fact represents `2.0`

I am a bit confused about how `substrate_fixed::Fixed(U/I)(32/64/128)` works compared to the vanilla substrate version.

Reply:
I agree in regards to `from_num` not being a constant function, now it's complicated to interpret the numbers.

The `U` type specifies the number of fractional bits. It is used to move the comma in the bit representation. For `FixedU32<U24>`, we have `32-24 = 8` of the highest bits for the integer part, and the remaining `24` bits for the fractional part. The lowest precision is `1.0 / pow(2, fractional_bits)`, so for `U24` this is `1.0 / 2.0^24.0 =  0.00000006`.

Reply:
Oh, fractions are bit related. Thanks for the explanation

Even if some cases aren't so straightforward, it is probably better to offset where possible, for example, `0x0200_0000` == `2 << 24`